compositor/sessionmanager/sessionmanager.cpp

```diff
@@ -320,9 +285,6 @@ void SessionManager::requestHybridSleep()
     Q_EMIT hybridSleepRequested();
 }
 
-void SessionManager::cancelShutdownRequest()
-{
-    Q_EMIT shutdownRequestCanceled();
-}
+void SessionManager::cancelShutdownRequest() { Q_EMIT shutdownRequestCanceled(); }
 
-#include "moc_sessionmanager.cpp"
+#include "sessionmanager.moc"
```

Comment:
So there's a Q_OBJECT in both the .h and now the .cpp. So what is happening is there's both a moc_sessionmanager.cpp and a sessionmanager.moc file:

```
$ ls compositor/*moc*
compositor/hawaii_automoc.cpp     compositor/moc_loginmanagerbackend.cpp  compositor/moc_screensaver.cpp          compositor/processlauncheradaptor.moc_parameters
compositor/moc_application.cpp    compositor/moc_loginmanager.cpp         compositor/moc_sessionmanager.cpp       compositor/sessionmanager.moc
compositor/moc_authenticator.cpp  compositor/moc_powermanagerbackend.cpp  compositor/moc_systemdpowerbackend.cpp
compositor/moc_fakebackend.cpp    compositor/moc_powermanager.cpp         compositor/moc_upowerpowerbackend.cpp
compositor/moc_logindbackend.cpp  compositor/moc_processlauncher.cpp      compositor/processlauncheradaptor.moc```

Here's where `moc_sessionmanager.cpp` is automatically being compiled:

```
$ cat compositor/hawaii_automoc.cpp   
/* This file is autogenerated, do not edit*/
#include "moc_fakebackend.cpp"
#include "moc_powermanagerbackend.cpp"
#include "moc_systemdpowerbackend.cpp"
#include "moc_upowerpowerbackend.cpp"
#include "moc_sessionmanager.cpp"
```

If I have both `moc_sessionmanager.cpp` and `sessionmanager.moc` being included at the bottom of the sessionmanager.cpp file, then `moc_sessionmanager is not included in hawaii_automoc.cpp`:

```
$  cat compositor/hawaii_automoc.cpp                                                                                   
/* This file is autogenerated, do not edit*/
#include "moc_fakebackend.cpp"
#include "moc_powermanagerbackend.cpp"
#include "moc_systemdpowerbackend.cpp"
#include "moc_upowerpowerbackend.cpp"
```

So, we must include the .moc file or the build will fail. We can optionally include moc_.cpp files, but since they're automatically compiled, I'd prefer not including them as its unnecessary.